src/3box.js

```diff
@@ -493,24 +497,29 @@ class Box {
   }
 
   async _linkProfile () {
-    let linkData = await this.public.get('ethereum_proof')
+    const address = this._3id.managementAddress
+    let linkData = await this._readAddressLink(address)
 
     if (!linkData) {
-      const address = this._3id.managementAddress
       const did = this._3id.muportDID
       let consent
       try {
         consent = await utils.getLinkConsent(address, did, this._web3provider)
       } catch (e) {
+        console.log(e)
         throw new Error('Link consent message must be signed before adding data, to link address to store')
       }
 
       linkData = {
-        consent_msg: consent.msg,
-        consent_signature: consent.sig,
-        linked_did: did
+        version: 1,
+        type: ACCOUNT_TYPES.ethereumEOA,
+        message: consent.msg,
+        signature: consent.sig,
+        timestamp: consent.timestamp
       }
-      await this.public.set('ethereum_proof', linkData, { noLink: true })
+
+      const docHash = (await this._ipfs.add(Buffer.from(JSON.stringify(linkData))))[0].hash
```

Comment:
I'd prefer if we use the DAG API instead of the files API: https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/DAG.md

Reply:
👍 yeah was going to change that